presto-main/src/test/java/com/facebook/presto/type/TestRealType.java

```diff
@@ -52,4 +56,18 @@ protected Object getGreaterValue(Object value)
         float greaterValue = intBitsToFloat(bits) + 0.1f;
         return Long.valueOf(floatToRawIntBits(greaterValue));
     }
+
+    @Test
+    public void testNanHash()
+    {
+        BlockBuilder blockBuilder = new IntArrayBlockBuilder(null, 4);
+        blockBuilder.writeInt(floatToIntBits(Float.NaN));
+        blockBuilder.writeInt(floatToRawIntBits(Float.NaN));
+        blockBuilder.writeInt(-0x400000);
```

Comment:
Same comment as above. If these are binary representations of NaN add a comment.